rust/oak/src/lib.rs

```diff
@@ -64,57 +65,31 @@ where
 
 type Handle = u64;
 
-// Keep in sync with /oak/server/oak_node.h.
-pub const LOGGING_CHANNEL_HANDLE: Handle = 1;
-pub const GRPC_IN_CHANNEL_HANDLE: Handle = 2;
-pub const GRPC_OUT_CHANNEL_HANDLE: Handle = 3;
-
-// Status values returned across the host function interface
-#[derive(Debug, PartialEq)]
-pub enum Status {
-    Ok,
-    BadHandle,
-    InvalidArgs,
-    ChannelClosed,
-    BufferTooSmall,
-    OutOfRange,
-    Unknown(i32),
-}
-
-fn status_from_i32(raw: i32) -> Status {
-    // Keep in sync with /oak/server/status.h
-    match raw {
-        0 => Status::Ok,
-        1 => Status::BadHandle,
-        2 => Status::InvalidArgs,
-        3 => Status::ChannelClosed,
-        4 => Status::BufferTooSmall,
-        5 => Status::OutOfRange,
-        _ => Status::Unknown(raw),
-    }
-}
-
-/// Map a host function status to the nearest available std::io::Result.
-fn result_from_status<T>(status: Status, val: T) -> std::io::Result<T> {
+/// Map an OakStatus to the nearest available std::io::Result.
+fn result_from_status<T>(status: Option<OakStatus>, val: T) -> std::io::Result<T> {
```

Comment:
Not this PR, but I find passing both status and value at the same time a bit awkward. Usually we should know whether we are handling an error or an actual value, and use an appropriate more precise method to convert if necessary? Anyway, we can try other approaches separately.

Reply:
The parameter has this type because the generated `OakStatus::from_i32(v: i32)` emits `-> Option<OakStatus>`.

(Aside: the proto-enum-derived code actually holds less information than the manual version did, because unknown status values lose their numeric value, but I reckoned it was still worthwhile to shift to something that automatically keeps in sync)

Reply:
How much do we care about converting our errors to `io::Error` anyways? I think we should just have our own `Result` type that uses `OakStatus` as error. WDYT?

Reply:
That might make sense, but I think the conversion would still be needed in the implementation of the `Write` trait for send channels.  (Dunno if we could drop that too, though.)